.github/workflows/frontend-lint-test-build.yaml

```diff
@@ -0,0 +1,266 @@
+name: Frontend
+on:
+  pull_request
+
+jobs:
+  # In order to be able to have required checks, a workflow can't be entirely
+  # skipped: see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks
+  paths-filter:
+    name: "Filter Paths"
+    runs-on: ubuntu-latest
+    outputs:
+      src: ${{ steps.filter.outputs.src }}
+    steps:
+    - uses: dorny/paths-filter@v3
+      id: filter
+      with:
+        filters: |
+          src:
+            - 'frontend/src/**'
+            - 'frontend/*.json'
+            - 'frontend/*.js'
+            - 'frontend/*.ts'
+            - '.github/workflows/frontend-lint-test-build.yaml'
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: paths-filter
+    if: needs.paths-filter.outputs.src == 'true'
+    steps:
+      # Setup:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
```

Comment:
Should we just bump this to 20 at least?

Reply:
Oh sure!